zerver/tests/test_users.py

```diff
@@ -2504,6 +2508,43 @@ def test_user_email_according_to_email_address_visibility_setting(self) -> None:
         self.assertEqual(result["user"].get("delivery_email"), hamlet.delivery_email)
         self.assertEqual(result["user"].get("email"), hamlet.delivery_email)
 
+    def test_restricted_access_to_users(self) -> None:
+        othello = self.example_user("othello")
+        cordelia = self.example_user("cordelia")
+        desdemona = self.example_user("desdemona")
+        realm = get_realm("zulip")
+
+        self.set_up_db_for_testing_user_access()
+
+        members_group = UserGroup.objects.get(name="role:members", realm=realm)
+        do_change_realm_permission_group_setting(
+            realm, "can_access_all_users_group", members_group, acting_user=None
+        )
+        self.login("polonius")
+        result = orjson.loads(self.client_get("/json/users").content)
```

Comment:
Let's capture the number of queries for this operation and assert it -- it's likely to be fairly expensive.

Reply:
Updated the test.